src-ghc/Liquid/GHC/GhcMonadLike.hs

```diff
@@ -294,14 +289,27 @@ lookupModule mod_name Nothing = do
       res <- findExposedPackageModule hsc_env mod_name Nothing
       case res of
         Found _ m -> return m
-        err       -> throwOneError $ noModError (hsc_dflags hsc_env) noSrcSpan mod_name err
-
--- Compatibility shim to extract the comments out of an 'ApiAnns', as modern GHCs now puts the
--- comments (i.e. Haskell comments) in a different field ('apiAnnRogueComments').
-apiComments :: ApiAnns -> [Ghc.Located AnnotationComment]
-apiComments apiAnns =
-  let comments = concat . M.elems . apiAnnComments $ apiAnns
-  in
-     map toRealSrc $ mappend comments (apiAnnRogueComments apiAnns)
+        err       ->
+          throwOneError $ noModError hsc_env noSrcSpan mod_name err
+
+-- | Abstraction of 'EpaComment'.
+data ApiComment
+  = ApiLineComment String
+  | ApiBlockComment String
+
+-- | Extract top-level comments from a module.
+apiComments :: ParsedModule -> [Ghc.Located ApiComment]
+apiComments pm =
```

Comment:
The behavior of this function has changed with the port.

For a module like
```
module A where

f :: Int
f = 1

{-@ ignore g @-}
g :: Int
g =2
```
before it would yield `{-@ ignore g @-}` but now it skips the comment (maybe because it is not attached as a module annotation?).

Reply:
I figured this one. A way to find missing comments is to use `ghc-exactprint` to print the AST with [showAst :: (Data a) => a -> String](https://hackage.haskell.org/package/ghc-exactprint-1.6.1.1/docs/Language-Haskell-GHC-ExactPrint.html#g:6).

This shows the whole tree, and in particular the node that holds the `{-@ ignore ... @-}` comment.

Now, I'm expecting this problem to come back when we depend on some comment attached to a function in a `where` clause. But I'm letting it come for now.